linera-views/src/journaling.rs

```diff
@@ -238,43 +237,37 @@ where
         base_key: &[u8],
     ) -> Result<JournalHeader, K::Error> {
         let mut iter = simplified_batch.into_iter();
-        let mut block = K::Batch::default();
+        let mut block_batch = K::Batch::default();
         let mut block_size = 0;
         let mut block_count = 0;
-        let mut transaction = K::Batch::default();
+        let mut transaction_batch = K::Batch::default();
         let mut transaction_size = 0;
-        loop {
-            if !iter.write_next_value(&mut block, &mut block_size)? {
-                break;
-            }
+        while iter.write_next_value(&mut block_batch, &mut block_size)? {
             let (block_flush, transaction_flush) = {
-                if iter.is_empty() || block.len() == K::MAX_BATCH_SIZE - 2 {
+                if iter.is_empty() || block_batch.len() == K::MAX_BATCH_SIZE - 2 {
                     (true, true)
                 } else {
-                    let next_block_size = iter.next_batch_size(&block, block_size)?;
+                    let next_block_size =
+                        block_size + iter.next_value_size()? + block_batch.overhead_size();
```

Comment:
Doesn't that use the overhead size _without_ the new element now? The vector lengths could have a longer encoding after being incremented by one.

Reply:
yes good catch